src/detect-template-buffer.c

```diff
@@ -48,6 +48,9 @@ static int DetectTemplateBufferSetup(DetectEngineCtx *de_ctx, Signature *s,
 {
     s->list = DETECT_SM_LIST_TEMPLATE_BUFFER_MATCH;
     s->alproto = ALPROTO_TEMPLATE;
+    s->flags |= SIG_FLAG_STATE_MATCH;
+    s->mask |= SIG_MASK_REQUIRE_TEMPLATE_STATE;
```

Comment:
I'm not convinced this is a good idea. The whole logic is currently all in src/detect.c, and scattering it over the keywords would not necessarily make it easier. Besides, I think if we decide to do it, we should do it for all and not just for this one. But for now I vote against it.